src/parser.ts

```diff
@@ -440,3 +402,62 @@ function getParamsOrders(ast: AST, tokenIndex: number) {
 
   return paramsOrder;
 }
+
+/**
+ * If the given tag has a default value, then this will add a note to the
+ * description with that default value. This is done because TypeScript does
+ * not display the documented JSDoc default value (e.g. `@param [name="John"]`).
+ *
+ * If the description already contains such a note, it will be updated.
+ */
+function addDefaultValueToDescription(
+  tag: commentParser.Tag,
+): commentParser.Tag {
+  if (tag.optional && tag.default) {
+    let { description } = tag;
+
+    // remove old note
+    description = description.replace(/[ \t]*Default is `.*`\.?$/, "");
+
+    // add a `.` at the end of previous sentences
+    if (description && !/[.\n]$/.test(description)) {
+      description += ".";
+    }
+
+    description += ` Default is \`${tag.default}\``;
+
+    return {
+      ...tag,
+      description: description.trim(),
+    };
+  } else {
+    return tag;
+  }
+}
+
+/**
+ * This will combine the `name`, `optional`, and `default` properties into name
+ * setting the other two to `false` and `undefined` respectively.
+ */
+function combineIntoName({
+  name,
+  optional,
+  default: default_,
+  ...rest
+}: commentParser.Tag): commentParser.Tag {
+  if (name && optional) {
+    // Figure out if tag type have default value
+    if (default_) {
+      name = `[${name}=${default_}]`;
+    } else {
+      name = `[${name}]`;
+    }
+  }
+
+  return {
+    ...rest,
+    name: name,
+    optional: false,
+    default: undefined,
+  };
```

Comment:
Don't remove optional and default, maybe in feature we needed it in another function, contributors can't find it.